src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMetadataSerializer.cs

```diff
@@ -121,6 +123,9 @@ protected virtual void ReadKeyDescriptorForSigning(WsFederationConfiguration con
                 KeyInfo keyInfo = new KeyInfo();
                 keyInfo.ReadFrom(reader);
                 configuration.KeyInfos.Add(keyInfo);
+
+                var cert = new X509Certificate2(Convert.FromBase64String(keyInfo.CertificateData));
```

Comment:
Need positive, negative tests.
For positive: You may want to create a WsFedComparer similar to: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/WsFed/test/Microsoft.IdentityModel.Xml.Tests/ReferenceXml.cs#L828
Then compare in you tests as in: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/WsFed/test/Microsoft.IdentityModel.Xml.Tests/DsigTests.cs#L136
You should create a WsFedTestSet similar to: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/WsFed/test/Microsoft.IdentityModel.Xml.Tests/ReferenceXml.cs#L821
For negative: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/WsFed/test/Microsoft.IdentityModel.Xml.Tests/ReferenceXml.cs#L568